### PR TITLE
(DOCSP -19163): Refactor Intro for Web Devs for Readability  

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Edit the web developer introduction page for readability. 

To assess readability used the tools:  
- https://hemingwayapp.com/
- http://readable.com/

Refactored so that it acheived score on Readable of 'B', per [direction in the ticket's epic](https://jira.mongodb.org/browse/DOCSP-19160?focusedCommentId=4175080&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4175080).

### Jira

- https://jira.mongodb.org/browse/DOCSP-19163

### Staged Changes (Requires MongoDB Corp SSO)

- [Introduction to MongoDB Realm for Web Developers](https://docs-mongodborg-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-19163/get-started/introduction-web)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/get-started/introduction-web.txt [hunk 1]
```diff
@@ -35,20 +29,16 @@ boilerplate backend components to writing the code that makes your app unique.
...
-
-MongoDB Realm provides built-in user management and authentication methods that
-encapsulate this complexity and make it easy to integrate third-party
-authentication providers like social media platforms and enterprise identity
-management systems. You can enable authentication providers with a simple
-server-side configuration and immediately log in from any client application
-using the Realm SDK.
+Secure user authentication is essential for most apps. Some apps authenticate
+users with traditional email/password combinations. Others use API keys or third-party
+authentication providers like Facebook and Google. These methods can get complicated,
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:34]: We shouldn't use "complicated" here, for similar reasons to "daunting" above. We might say something adds complexity if it adds a lot of additional code, or may be difficult to reason about logically. What if we instead reword this to something like this that provides a more specific consequence:

> Implementing these auth providers may add a lot of boilerplate code to your app. This added complexity expands the surface area for potential bugs.

File: source/get-started/introduction-web.txt [hunk 2]
```diff
@@ -35,20 +29,16 @@ boilerplate backend components to writing the code that makes your app unique.
...
+and often require a lot of boilerplate code.
+ 
+MongoDB Realm provides built-in user management and authentication methods. These
+features encapsulate complexity, and make it easy to integrate third-party authentication
+providers. You can enable authentication providers with a simple server-side configuration.
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:39]: We should avoid saying "simple" here for similar reasons as above. 
```suggestion
providers. You can enable authentication providers in your server-side configuration.
```

File: source/get-started/introduction-web.txt [hunk 3]
```diff
@@ -35,20 +29,16 @@ boilerplate backend components to writing the code that makes your app unique.
...
+Then immediately log in from any client application using a Realm SDK.
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:40]:
```suggestion
Then immediately log in from any client application using a {+backend-short+} SDK.
```

File: source/get-started/introduction-web.txt [hunk 4]
```diff
@@ -4,25 +4,19 @@ Introduction to MongoDB Realm for Web Developers
...
-access and security rules. This allows you to shift your focus from implementing
-boilerplate backend components to writing the code that makes your app unique.
+Modern web applications are more powerful than ever before. Frontend technologies 
+like React and Angular let developers create web apps that behave like native applications. 
+However, as the use of web applications grows, so do the demands of maintaining them.
+
+Developers traditionally need to serve data to the browser from a server with a 
+RESTful API. Developers also must secure the server with features like authentication
+and data validation. A wide variety of libraries and frameworks exist to address these scenarios.
+But deciding the set of solutions with the right tradeoffs for your use case can be daunting.
```
Comment (cbush) [inline source/get-started/introduction-web.txt:14]:
```suggestion
But deciding on the set of solutions with the right tradeoffs for your use case can be daunting.
```

File: source/get-started/introduction-web.txt [hunk 5]
```diff
@@ -57,24 +47,22 @@ To learn more about authentication in MongoDB Realm, see :doc:`Users & Authentic
...
-Realm secures data with role-based data access and manipulation rules that you
-configure for all data that your application uses. These dynamically determine
-each user's access to every document for every request, which means that a user
-can only ever insert, read, or modify data if both they and the document meet
-pre-defined conditions.
+Modern applications require that data is available using consistent types and formats.
+Data consistency guarantees that all the application's components can work with it.
+It's also important to make sure that any given piece of data is only accessible
+to an authorized user.  For example, you might only want to let a user have access
+to their own data.
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:54]: We could move the "only" here to be more explicit, and we could also imply the active permission process with a slight rewording:

```suggestion
to an authorized user.  For example, you might grant a user access
to only their own data.
```

File: source/get-started/introduction-web.txt [hunk 6]
```diff
@@ -35,20 +29,16 @@ boilerplate backend components to writing the code that makes your app unique.
...
-management systems. You can enable authentication providers with a simple
-server-side configuration and immediately log in from any client application
-using the Realm SDK.
+Secure user authentication is essential for most apps. Some apps authenticate
+users with traditional email/password combinations. Others use API keys or third-party
+authentication providers like Facebook and Google. These methods can get complicated,
+and often require a lot of boilerplate code.
+ 
+MongoDB Realm provides built-in user management and authentication methods. These
+features encapsulate complexity, and make it easy to integrate third-party authentication
```
Comment (cbush) [inline source/get-started/introduction-web.txt:38]: Also: could "encapsulate complexity" be simplified?
```suggestion
features encapsulate complexity and make it easy to integrate third-party authentication
```

File: source/get-started/introduction-web.txt (hunk 3 above)
Comment (cbush) [inline source/get-started/introduction-web.txt:40]: This doesn't make sense as a new sentence. Shouldn't it be:

>...configuration then immediately...

File: source/get-started/introduction-web.txt [hunk 7]
```diff
@@ -57,24 +47,22 @@ To learn more about authentication in MongoDB Realm, see :doc:`Users & Authentic
...
-`{+atlas+} <https://www.mongodb.com/realm?tck=docs_realm#atlas-form-container>`_
-collection to data in your web app.
-
-Realm secures data with role-based data access and manipulation rules that you
-configure for all data that your application uses. These dynamically determine
-each user's access to every document for every request, which means that a user
-can only ever insert, read, or modify data if both they and the document meet
-pre-defined conditions.
+Modern applications require that data is available using consistent types and formats.
+Data consistency guarantees that all the application's components can work with it.
```
Comment (cbush) [inline source/get-started/introduction-web.txt:51]: Work with what?

File: source/get-started/introduction-web.txt [hunk 8]
```diff
@@ -57,24 +47,22 @@ To learn more about authentication in MongoDB Realm, see :doc:`Users & Authentic
...
+to an authorized user.  For example, you might only want to let a user have access
+to their own data.
+
+MongoDB Realm validates data with a Realm schema you define in the `JSON schema <https://json-schema.org/>`__ standard.
+A Realm schema serves as the source of truth for defining data types in your application.
+Realm also uses the schema to map data between your application and a 
+`{+atlas+} <https://www.mongodb.com/realm?tck=docs_realm#atlas-form-container>`_ collection.
+
+You can also configure Realm to secure data with role-based data access rules.
+These rules dynamically determine each user's access to every document for every
```
Comment (cbush) [inline source/get-started/introduction-web.txt:62]: Do we need the word "dynamically" here? Aside from assembling an annoying alliteration with "determine", it feels redundant when we already talk about "every request" later.

File: source/get-started/introduction-web.txt [hunk 9]
```diff
@@ -84,17 +72,16 @@ To learn more about defining a schema and access rules for your data, see
...
-Many applications require that some business logic runs on the app server
-instead of in the browser. Usually this logic deals with sensitive user data or
-coordinates data from multiple external services. Apps might also need to
-respond automatically in real time when something happens in the app or in an
-external service.
+Many applications require that some business logic runs on a server. Usually this
+logic deals with sensitive user data or interacting with external services. Apps might
+also need to respond in real time when something happens in the app or in an external service.
+
+MongoDB Realm lets you define serverless JavaScript functions called Realm Functions.
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:79]:
```suggestion
{+backend+} lets you define serverless JavaScript functions called {+backend-short+} Functions.
```

File: source/get-started/introduction-web.txt [hunk 10]
```diff
@@ -57,24 +47,22 @@ To learn more about authentication in MongoDB Realm, see :doc:`Users & Authentic
...
-each user's access to every document for every request, which means that a user
-can only ever insert, read, or modify data if both they and the document meet
-pre-defined conditions.
+Modern applications require that data is available using consistent types and formats.
+Data consistency guarantees that all the application's components can work with it.
+It's also important to make sure that any given piece of data is only accessible
+to an authorized user.  For example, you might only want to let a user have access
+to their own data.
+
+MongoDB Realm validates data with a Realm schema you define in the `JSON schema <https://json-schema.org/>`__ standard.
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:56]:
```suggestion
{+backend+} validates data with a {+backend-short+} schema you define in the `JSON schema <https://json-schema.org/>`__ standard.
```

File: source/get-started/introduction-web.txt [hunk 11]
```diff
@@ -57,24 +47,22 @@ To learn more about authentication in MongoDB Realm, see :doc:`Users & Authentic
...
+A Realm schema serves as the source of truth for defining data types in your application.
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:57]:
```suggestion
A {+backend-short+} schema serves as the source of truth for defining data types in your application.
```

File: source/get-started/introduction-web.txt [hunk 12]
```diff
@@ -57,24 +47,22 @@ To learn more about authentication in MongoDB Realm, see :doc:`Users & Authentic
...
+Realm also uses the schema to map data between your application and a
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:58]:
```suggestion
{+backend-short+} also uses the schema to map data between your application and a 
```

File: source/get-started/introduction-web.txt [hunk 13]
```diff
@@ -57,24 +47,22 @@ To learn more about authentication in MongoDB Realm, see :doc:`Users & Authentic
...
+It's also important to make sure that any given piece of data is only accessible
+to an authorized user.  For example, you might only want to let a user have access
+to their own data.
+
+MongoDB Realm validates data with a Realm schema you define in the `JSON schema <https://json-schema.org/>`__ standard.
+A Realm schema serves as the source of truth for defining data types in your application.
+Realm also uses the schema to map data between your application and a 
+`{+atlas+} <https://www.mongodb.com/realm?tck=docs_realm#atlas-form-container>`_ collection.
+
+You can also configure Realm to secure data with role-based data access rules.
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:61]:
```suggestion
You can also configure {+backend-short+} to secure data with role-based data access rules.
```

File: source/get-started/introduction-web.txt [hunk 14]
```diff
@@ -104,58 +91,41 @@ To learn more about defining and using serverless functions, see :doc:`Functions
...
-for which they have permission.
+Web developers have to access their data efficiently and securely. 
+If the data is on a server, you need an API to access the data and prevent unauthorized access.
+
+The Realm Web SDK includes a :ref:`secure MongoDB client <intro-web-mql>` that uses
+:ref:`GraphQL <intro-web-graphql>` and the :manual:`Query API </tutorial/query-documents>`
+to access data in {+atlas+}. Use either or both of these APIs to work with
+your application's data.
+
+Realm's server-side rules ensures that users only read and modify permitted document.
```
Comment (cbush) [inline source/get-started/introduction-web.txt:102]:
```suggestion
Realm's server-side rules ensures that users only read and modify permitted documents. 
```

Comment (cbush) [reply source/get-started/introduction-web.txt:102]: I might say "documents they have access to" or something, as "permitted documents" sounds more obscure.

**Review by cbush — COMMENTED** [review]
Much improved!

File: source/get-started/introduction-web.txt [hunk 15]
```diff
@@ -104,58 +91,41 @@ To learn more about defining and using serverless functions, see :doc:`Functions
...
-
-Realm secures data with server-side rules that dynamically return and withhold
-data depending on the user that issued a query. Rules consist of one or more
-user roles for each collection. A role determines whether a given user has
-permission to insert or modify a document as well as if they can read or write
-specific fields in the document. Realm dynamically determines a role for each
-document in every request and enforces the role's permissions for its
-corresponding document. This ensures that users only read and modify documents
-for which they have permission.
+Web developers have to access their data efficiently and securely.
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:94]: This read a little awkwardly to me, with "developers have to" and then "you" in the next sentence. Maybe we could reword a bit so it flows more naturally; something like:

> Developers need efficient, secure access to data.

File: source/get-started/introduction-web.txt [hunk 16]
```diff
@@ -104,58 +91,41 @@ To learn more about defining and using serverless functions, see :doc:`Functions
...
-user roles for each collection. A role determines whether a given user has
-permission to insert or modify a document as well as if they can read or write
-specific fields in the document. Realm dynamically determines a role for each
-document in every request and enforces the role's permissions for its
-corresponding document. This ensures that users only read and modify documents
-for which they have permission.
+Web developers have to access their data efficiently and securely. 
+If the data is on a server, you need an API to access the data and prevent unauthorized access.
+
+The Realm Web SDK includes a :ref:`secure MongoDB client <intro-web-mql>` that uses
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:97]: We could split this a little more to improve readability a little further:

```suggestion
The Realm Web SDK includes a :ref:`secure MongoDB client <intro-web-mql>` to interact with data. It uses
```

File: source/get-started/introduction-web.txt [hunk 17]
```diff
@@ -104,58 +91,41 @@ To learn more about defining and using serverless functions, see :doc:`Functions
...
+:ref:`GraphQL <intro-web-graphql>` and the :manual:`Query API </tutorial/query-documents>`
+to access data in {+atlas+}. Use either or both of these APIs to work with
+your application's data.
+
+Realm's server-side rules ensures that users only read and modify permitted document. 
+Rules consist of one or more user roles for each collection. A role determines if
+a given user has permission to access and edit a document. Realm determines a role
+for each document in every request and enforces the role's permissions before responding.
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:105]: Could split to improve readability:
```suggestion
for each document in every request. It enforces the role's permissions before responding.
```

**Review by dacharyc — COMMENTED** [review]
This is a huge improvement - nice work! I've made a handful of suggestions here that could clarify details, improve flow, or remove language that doesn't really seem to fit. I also started suggesting source constants, but gave up partway through because there are so many instances of Realm and it was turning into so many comments. Feel free to implement source constants here or not!

File: source/get-started/introduction-web.txt (hunk 17 above)
Comment (mongodben) [reply source/get-started/introduction-web.txt:105]: adding `then` to make clear it's sequential: 

```suggestion
for each document in every request. It then enforces the role's permissions before responding.
```

File: source/get-started/introduction-web.txt (hunk 3 above)
Comment (mongodben) [reply source/get-started/introduction-web.txt]: yea you're right. i prob broke up sentence to appease the complexity analyzer tool. fixing

File: source/get-started/introduction-web.txt (hunk 8 above)
Comment (mongodben) [reply source/get-started/introduction-web.txt:62]: i like alliteration 😄. but removing b.c unnec here

File: source/get-started/introduction-web.txt (hunk 4 above)
Comment (mongodben) [reply source/get-started/introduction-web.txt:14]: i wasn't convinced of this change at first. but after some additional research, you're def right. adding in next batch of changes.

ref: https://english.stackexchange.com/a/62843

File: source/get-started/introduction-web.txt (hunk 7 above)
Comment (mongodben) [reply source/get-started/introduction-web.txt:51]: idk..poor phrasing. changing to "work together"

File: source/get-started/introduction-web.txt [hunk 18]
```diff
@@ -4,25 +4,19 @@ Introduction to MongoDB Realm for Web Developers
...
-
-MongoDB Realm is a serverless application backend that streamlines these common
-challenges with configurable functions and services that integrate Realm's data
-access and security rules. This allows you to shift your focus from implementing
-boilerplate backend components to writing the code that makes your app unique.
+Modern web applications are more powerful than ever before. Frontend technologies 
+like React and Angular let developers create web apps that behave like native applications. 
+However, as the use of web applications grows, so do the demands of maintaining them.
+
+Developers traditionally need to serve data to the browser from a server with a
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/get-started/introduction-web.txt:11]: REST is def [a standard practice](https://www.postman.com/state-of-api/api-technologies/#api-technologies), but we can sidestep mentioning it here by just removing the adjective and saying API.

File: source/get-started/introduction-web.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (mongodben) [reply source/get-started/introduction-web.txt:14]: i like the change. edited slightly to make clear that it's not *just* this security stuff.

File: source/get-started/introduction-web.txt [hunk 19]
```diff
@@ -4,25 +4,19 @@ Introduction to MongoDB Realm for Web Developers
...
-respond to incoming requests. You can solve each of these problems in isolation
-with a wide variety of libraries and frameworks, but deciding which solution
-solves each problem with the right tradeoffs for your use case can be a daunting
-challenge.
-
-MongoDB Realm is a serverless application backend that streamlines these common
-challenges with configurable functions and services that integrate Realm's data
-access and security rules. This allows you to shift your focus from implementing
-boilerplate backend components to writing the code that makes your app unique.
+Modern web applications are more powerful than ever before. Frontend technologies
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/get-started/introduction-web.txt:7]: i see your point. i just cut the paragraph and tweaked the subsequent one. 

as MongoDB Realm is a **backend** as a service, we don't need to dwell on talking about the frontend.

**Comment by mongodben** [comment]
@dacharyc and @cbush thanks for the feedback here! ready for re-review

File: source/get-started/introduction-web.txt [hunk 20]
```diff
@@ -4,25 +4,16 @@ Introduction to MongoDB Realm for Web Developers
...
-with a wide variety of libraries and frameworks, but deciding which solution
-solves each problem with the right tradeoffs for your use case can be a daunting
-challenge.
-
-MongoDB Realm is a serverless application backend that streamlines these common
-challenges with configurable functions and services that integrate Realm's data
-access and security rules. This allows you to shift your focus from implementing
-boilerplate backend components to writing the code that makes your app unique.
+Building a web app usually requires sending data from a server to the browser.
+This comes with all kinds of overhead, such as authentication, and data validation.
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:8]: This paragraph has a few phrases which, when used together, make it feel too informal. "usually "all kinds of" "And then" here combine to make the tone feel a little off compared to the rest of our docs. We could reword slightly to improve. I think we could drop "usually" and "all kinds of" if you want to keep the "And then" - although I would probably eliminate all three.

File: source/get-started/introduction-web.txt [hunk 21]
```diff
@@ -4,25 +4,16 @@ Introduction to MongoDB Realm for Web Developers
...
-access and security rules. This allows you to shift your focus from implementing
-boilerplate backend components to writing the code that makes your app unique.
+Building a web app usually requires sending data from a server to the browser.
+This comes with all kinds of overhead, such as authentication, and data validation. 
+And then the application has to be scalable and secure. 
+Many libraries and frameworks exist to provide these services.
+Selecting the best ones for your project requires deep understanding of the tradeoffs.
+
+{+backend+} is a serverless application backend that streamlines solving these 
+common challenges. {+backend-short+} comes with configurable functions, integrated data access, and security rules.
```
Comment (dacharyc) [inline source/get-started/introduction-web.txt:14]: This is a total nit, but I think we could reword here to make our point stronger. "Comes with" sounds like "When you use Realm, you also get these things on the side" - but really, we want to point out that Realm solves the problems you outlined above by "providing" or "offering" these features. This is a subtle nuance and feel free to take it or leave it, but I just wanted to point it out.
 
```suggestion
common challenges. {+backend-short+} provides configurable functions, integrated data access, and security rules.
```

File: source/get-started/introduction-web.txt (hunk 1 above)
Comment (dacharyc) [reply source/get-started/introduction-web.txt:31]: I really do think we should remove the "complicated" here. We're establishing this problem to highlight Realm as a solution or counterpoint, and using "complicated" here implies that Realm is _not_ "complicated" when we propose it as a solution in the next paragraph. I've been through our auth provider procedures and code examples, and I don't think I'd imply that it's not "complicated" with us.

**Review by dacharyc — APPROVED** [review]
Great work, Ben! I've left a few comments about some lingering nits, but it's getting pretty subjective at this point so feel free to take it or leave it.

This is a big improvement to the page's readability, and I think it also positions Realm's offerings more strongly - nice job!

File: source/get-started/introduction-web.txt (hunk 20 above)
Comment (mongodben) [reply source/get-started/introduction-web.txt:8]: refactored a bit. kept usually for parallelism of ideas w the following paragraph 1st sentence which talks abt "common challenges".
